### PR TITLE
[Forwardport] [BUGFIX] Correct GitHub Linkhandler

### PR DESCRIPTION
**Pull request by linawolf**
Forward port from https://github.com/TYPO3-Documentation/t3docs-examples/pull/132

The TypolinkBuilder was missing all together. This example can never have worked as in promised in the docs

releases: main

### CODE REVIEW

**Review by brotkrueml — COMMENTED** [review]
Should the title be "for TYPO3 v12" instead of "for TYPO3 v11"?

**Comment by linawolf** [comment]
It is a preport of the solution for v11, v12 is not solved completely yet

**Comment by brotkrueml** [comment]
> It is a preport of the solution for v11, v12 is not solved completely yet

Ah, then the title is misleading. Anyway, as I made some changes to the v11 branch solution, these should be merged also into here?

**Comment by linawolf** [comment]
> > It is a preport of the solution for v11, v12 is not solved completely yet
> 
> Ah, then the title is misleading. Anyway, as I made some changes to the v11 branch solution, these should be merged also into here?

Could you propose a better title?

**Comment by brotkrueml** [comment]
> > > It is a preport of the solution for v11, v12 is not solved completely yet
> > 
> > 
> > Ah, then the title is misleading. Anyway, as I made some changes to the v11 branch solution, these should be merged also into here?
> 
> Could you propose a better title?

Well, as we have "backport" sometimes in the title what about "[Forwardport] [BUGFIX] Correct GitHub Linkhandler" with a description: "Forwardport from <the-commithash>"

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 - Use ContextualFeedbackSeverity for flash message severity (#87)
 
 ### Fixed
+- Fix the Github Linkhandler for TYPO3 v11 (#133)
```
Comment (brotkrueml) [inline CHANGELOG.md:30]: This does not apply to main/v12-compatible version?

File: Classes/LinkHandler/GithubLinkBuilder.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+use TYPO3\CMS\Frontend\Typolink\UnableToLinkException;
+
+/**
+ * Builds a TypoLink to a Github issue
+ */
+class GithubLinkBuilder extends AbstractTypolinkBuilder
+{
+    const TYPE_GITHUB = 'github';
+
+    /**
```
Comment (brotkrueml) [inline Classes/LinkHandler/GithubLinkBuilder.php:32]: Can we get rid of this docblock? It adds no value.

File: Classes/LinkHandler/GithubLinkBuilder.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        array $conf
+    ): LinkResultInterface {
+        $issueId = trim($linkDetails['value']);
+        if (str_starts_with($issueId, '#')) {
+            $issueId = substr($issueId, 1);
+        }
+        $issueId = (int)$issueId;
+        if ($issueId < 1) {
+            throw new UnableToLinkException(
+                '"' . $linkDetails['value'] . '" is not a valid Github issue number.',
```
Comment (brotkrueml) [inline Classes/LinkHandler/GithubLinkBuilder.php:48]:
```suggestion
                '"' . $linkDetails['value'] . '" is not a valid GitHub issue number.',
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]
Also the additional changed from the review process should be taken into account.

File: Classes/LinkHandler/GithubLinkBuilder.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ */
+class GithubLinkBuilder extends AbstractTypolinkBuilder
+{
+    const TYPE_GITHUB = 'github';
+
+    /**
+     * @inheritdoc
+     */
+    public function build(
+        array &$linkDetails,
```
Comment (linawolf) [inline Classes/LinkHandler/GithubLinkBuilder.php:36]:
```suggestion

    public function build(
        array &$linkDetails,
```

File: CHANGELOG.md (hunk 1 above)
Comment (linawolf) [inline CHANGELOG.md:30]:
```suggestion
### Fixed
```

**Review by brotkrueml — COMMENTED** [review]
I am getting the error

     Class "T3docs\\Examples\\LinkHandler\\GitHubLinkHandler" not found

when opening the link browser. Looked into it but haven't found where it is configured.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**